File: documentation-examples-java/src/java/dev/typr/foundations/docs/core/TransactorConfig.java
package dev.typr.foundations.docs.core;

import dev.typr.foundations.Transactor;
import dev.typr.foundations.connect.PgConfig;

@SuppressWarnings("unused")
public class TransactorConfig {
  PgConfig config = PgConfig.builder("localhost", 5432, "mydb", "user", "pass").build();
  // start
  Transactor tx = Transactor.create(config);
  Transactor testTx = tx.rollbackOnly();
  // stop
}

File: documentation-examples-kotlin/src/kotlin/dev/typr/foundationskt/docs/core/TransactorConfig.kt
package dev.typr.foundationskt.docs.core

import dev.typr.foundationskt.*
import dev.typr.foundationskt.connect.*

@Suppress("unused")
class TransactorCustomStrategy {
    val config = PgConfig.builder("localhost", 5432, "mydb", "user", "pass").build()

    //start
    val tx: Transactor = Transactor.create(config)

    val testTx: Transactor = tx.rollbackOnly()

    val txWithListener: Transactor = tx.withListener(QueryListener.NOOP)
    //stop
}
